### DISCUSSION TITLE
高级版统计（用于英文阅读）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1590] [插件说明]
描述
---

> 针对需要做英语阅读的用户，功能包括：**当前页面字符** **汉字字符** **段落数** **英文单词** **阅读进度** **阅读时间** 这几个维度来辅助你的英文阅读。

地址
---

[https://simpread.ksria.cn/plugins/details/3PHAZerSkb](https://simpread.ksria.cn/plugins/details/3PHAZerSkb)

截图
---

![image](https://user-images.githubusercontent.com/81074/133388457-e1071e90-105d-49d4-a441-b61a2ebdd59c.png)

选项
---

![image](https://user-images.githubusercontent.com/81074/133388186-972a2f01-232e-4f8f-a10f-de5eacd9a113.png)

> 当进入阅读模式后是否自动加载？

![image](https://user-images.githubusercontent.com/81074/133389178-855a856e-028c-40e6-8101-276f65eb7dcc.png)

> 选否后，可以在**阅读模式 →  控制栏 → 动作 → 插件触发器**手动进入

![image](https://user-images.githubusercontent.com/81074/133390541-0fac1d27-ffce-4e71-a5a9-060122a4bf41.png)

> 是否显示 mini 模式

![image](https://user-images.githubusercontent.com/81074/133389436-3e6c920b-0a14-4d02-951a-5d72f727d342.png)

> 选中后默认统计信息全部缩进，鼠标移上时显示

![image](https://user-images.githubusercontent.com/81074/133389544-7eadc529-a435-49b6-80dc-0b85c972f034.png)

![image](https://user-images.githubusercontent.com/81074/133389576-11f0ddc3-a466-4e54-a780-aa5af68aa9e2.png)

> **当前页面字符** **汉字字符** **段落数** **英文单词** **阅读进度** **阅读时间** 的定制化

![image](https://user-images.githubusercontent.com/81074/133389364-9561dc49-37a2-4e84-8cab-f7052b71c368.png)

更新日志
---

### 1.0.0

- `Optimize` 样式

- 设置

    - 进入阅读模式后是否运行？

    - 是否使用短栏模式？鼠标以上时自动展开

    - 是否显示字符统计 / 汉字字符统计 / 段落统计 / 英文单词统计 / 阅读进度 / 阅读时间？

### 0.0.3

- 完成 **当前页面字符** **汉字字符** **段落数** **英文单词** **阅读进度** **阅读时间** 的显示

关联
---

> 如果用于英文学习和阅读话，可以辅助使用以下方案：

- [全文翻译](https://simpread.ksria.cn/plugins/details/Y7JxbP7B4H)

- [适配扇贝阅读的适配站点](https://github.com/Kenshin/simpread/discussions/2705)

- [英文阅读版面辅助增强](https://simpread.ksria.cn/plugins/details/VGUmFrRs2b)

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by ydprtm** [comment]
才发现，一开始就装了这个插件了，但是看英文内容时，从来没有出现过最下边的统计功能条啊？

**Reply by Kenshin** [reply]
嗨，我测试了下，没有问题的，请看下图

![image](https://user-images.githubusercontent.com/81074/127654522-5fc22c35-32ac-43b9-b5e5-9bfed64b86a2.png)


另外，请确保你已成功安装了插件，位置在：选项页 → 插件管理，里面确保存在此插件

![image](https://user-images.githubusercontent.com/81074/127654638-f1b52f67-ac75-4d95-9b84-046ae076d4d9.png)

**Reply by ydprtm** [reply]
又安装了一个才发现的需要的“删除描述”插件，然后又恢复了触发器，再用官方适配的 medium.com 试了试，还是没有啊？

https://user-images.githubusercontent.com/822076/128629047-1151b017-ef7d-4689-ba45-6abe580563b2.mp4

**Reply by Kenshin** [reply]
安装插件后，请仔细看插件的描述 https://github.com/Kenshin/simpread/discussions/2620

**Comment by ydprtm** [comment]
提 `“删除描述”插件` 只是为了说明 又经历了一次 配置文件 的折腾来 `恢复触发器`，因此插件的安装、配置应该都没问题了。

专门看了说明，medium.com 是`官方适配`了的，又挑了个尽量长的文章来测试。

但是，还是没有这个`高级版统计`的状态条啊，不管是直接进入的 阅读页面，还是稍后读里……

**Reply by Kenshin** [reply]
emmm 确实很奇怪。你试试其它的页面，随便找个 https://sspai.com/post/67074

**Reply by ydprtm** [reply]
少数派的页面同样没有这个高级统计。

根据我的理解，如果这个插件是工作的，那么我进入阅读模式，然后用开发者工具在源代码中查找，应该是能搜索到那些统计显示用的词的（“进度”，“单词”）——但是在少数派这个页面测试的结果是没有任何结果，应该是插件没有工作，或者被屏蔽了？但是我没有任何广告屏蔽或者样式修改插件……

**Reply by Kenshin** [reply]
确实奇怪，你要是原意折腾的话，可以这么测试：

> 测试前一定要导出配置文件到本地


1. 重新安装简悦
2. 仅安装这个插件
3. **隐身模式下仅使用简悦的方式** 进入 https://sspai.com/post/67074
4. 左下角是否存在这个插件？

它在 Devetools 里面的应该是这样的

![image](https://user-images.githubusercontent.com/81074/130345344-d579975b-23f9-4851-8225-428cb21b488e.png)

**Reply by byheaven** [reply]
出现了同样的问题，实际上不止是这一个插件，其他例如小白点那个也不行，经过详细测试发现应该是插件通过配置文件同步和读取的问题，在开启自动同步后，安装插件的内容并不会同步到本地文件(与之相反，选择“清除本地插件”反而能触发主动同步)。感觉应该是同步助手的bug吧？

**Reply by Kenshin** [reply]
> 出现了同样的问题，实际上不止是这一个插件，其他例如小白点那个也不行，经过详细测试发现应该是插件通过配置文件同步和读取的问题，在开启自动同步后，安装插件的内容并不会同步到本地文件(与之相反，选择“清除本地插件”反而能触发主动同步)。感觉应该是同步助手的bug吧？

这是个已知的 Bug，跟 @ydprtm 描述的问题无关，解决方案 https://github.com/Kenshin/simpread/discussions/2342

**Comment by hao-lee** [comment]
当标签页不处于焦点时可不可以停止计时，比如我切到别的标签了或窗口了。

**Reply by Kenshin** [reply]
emmm 可以考虑，但可能比较「麻烦」

最好的办法的：手动方式（加个 暂停/继续 的功能）

**Reply by hao-lee** [reply]
> emmm 可以考虑，但可能比较「麻烦」
> 
> 最好的办法的：手动方式（加个 暂停/继续 的功能）

也是不错的办法

**Reply by Kenshin** [reply]
我先记录下需求

**Reply by hao-lee** [reply]
多谢~
